Dockerfile

```diff
@@ -0,0 +1,56 @@
+# This image is really a _starting point_. Some paths for improvements:
+#
+# - Produce a static linux binary using musl
+# - Find a way to split the build in multiple steps which can be cached (by Docker) better. Possibly not using Nix.
```

Comment:
What's the problem with caching right now? Do you mean better caching in a way that the three image (tags) do share more layers?